execute/factory.go

```diff
@@ -115,7 +119,7 @@ func (p PluginFactory) NewReportingPlugin(
 			p.execCodec,
 			p.msgHasher,
 			p.homeChainReader,
-			nil, // TODO: token data reader
+			p.tokenDataReader,
 			p.estimateProvider,
```

Comment:
estimate provider is before data reader in the above code.

Reply:
Fixed